sdk/synapse/synapse-spark/review/synapse-spark.api.md

```diff
@@ -0,0 +1,525 @@
+    };
+};
+
+// @public
+export interface GetSparkBatchJobOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+}
+
+// @public
+export type GetSparkBatchJobResponse = SparkBatchJob & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkBatchJob;
+    };
+};
+
+// @public
+export interface GetSparkSessionOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+}
+
+// @public
+export type GetSparkSessionResponse = SparkSession & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkSession;
+    };
+};
+
+// @public (undocumented)
+export type GetSparkStatementOptions = OperationOptions;
+
+// @public
+export type GetSparkStatementResponse = SparkStatement & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkStatement;
+    };
+};
+
+// @public (undocumented)
+export interface ListPageSettings {
+    continuationToken?: string;
+}
+
+// @public
+export interface ListSparkBatchJobsOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+    fromParam?: number;
+    size?: number;
+}
+
+// @public
+export type ListSparkBatchJobsResponse = SparkBatchJobCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkBatchJobCollection;
+    };
+};
+
+// @public
+export interface ListSparkSessionsOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+    fromParam?: number;
+    size?: number;
+}
+
+// @public
+export type ListSparkSessionsResponse = SparkSessionCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkSessionCollection;
+    };
+};
+
+// @public (undocumented)
+export type ListSparkStatementsOptions = OperationOptions;
+
+// @public
+export type ListSparkStatementsResponse = SparkStatementCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkStatementCollection;
+    };
+};
+
+// @public
+export const logger: import("@azure/logger").AzureLogger;
+
+// @public (undocumented)
+export interface OperationResponse {
+    _response: HttpResponse;
+}
+
+export { PipelineOptions }
+
+// @public
+export type PluginCurrentState = "Preparation" | "ResourceAcquisition" | "Queued" | "Submission" | "Monitoring" | "Cleanup" | "Ended" | string;
+
+// @public (undocumented)
+export type ResetSparkSessionTimeoutOptions = OperationOptions;
+
+// @public
+export type SchedulerCurrentState = "Queued" | "Scheduled" | "Ended" | string;
+
+// @public (undocumented)
+export interface SparkBatchJob {
+    appId?: string;
+    appInfo?: {
+        [propertyName: string]: string;
+    };
+    artifactId?: string;
+    errors?: SparkServiceError[];
+    id: number;
+    jobType?: SparkJobType;
+    // (undocumented)
+    livyInfo?: SparkBatchJobState;
+    logLines?: string[];
+    name?: string;
```

Comment:
is `id` really the only required property? I'd think maybe name and state and some others might be required.

Reply:
Thanks for catching it, I will check with backend.

---
In reply to: [521608988](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521608988) [](ancestors = 521608988)

Reply:
We plan to fix it when board review, that is a common issue between all languages.

---
In reply to: [521874970](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521874970) [](ancestors = 521874970,521608988)

Reply:
The board review is generally done by looking at the APIView of the packages you have created as well as code samples that use these packages, so I'm not sure what you mean by this?